src/main/kotlin/uk/gov/dluhc/printapi/messaging/MessageQueue.kt

```diff
@@ -1,12 +1,24 @@
 package uk.gov.dluhc.printapi.messaging
 
 import io.awspring.cloud.messaging.core.QueueMessagingTemplate
+import org.springframework.messaging.Message
+import org.springframework.messaging.core.MessagePostProcessor
 
 class MessageQueue<T : Any>(
     private val queueName: String,
-    private val queueMessagingTemplate: QueueMessagingTemplate
+    private val queueMessagingTemplate: QueueMessagingTemplate,
 ) {
 
-    fun submit(message: T) =
-        queueMessagingTemplate.convertAndSend(queueName, message)
+    fun submit(payload: T) =
```

Comment:
This is the only bit that is a perhaps a little .... eyebrow raising ....

Previously we were using `convertAndSend` on `QueueMessagingTemplate`. This calls a handful of methods in `QueueMessageTemplate` to convert the payload into a `Message`, then send the `Message` via the `send` method.

We need our aspect to get hold of the `Message`, so we have a pointcut over the `send` method (in the abstract class, but for all intents and purposes its a pointcut over `QueueMessageTemplate#send`

This didn't work :( It turns out that it doesn't work because of the proxy that AOP creates, in that you can only weave over methods that are called directly. You cant weave over methods that are called "internally" by the proxy! 

Kinda makes sense, but has forced me to change our class here so that it calls `QueueMessageTemplate.send`

Which takes me onto the next eyebrow raising part ...

In order to call `send` we need a `Message`. To create a message with all the correct headers, and to use the correct message converters etc we want to be able to call `QueueMessageTemplate.doConvert` .... but it's `protected`, so we can't. A cheeky extension function with some reflection hum-dingery 😁 to the rescue!
(or some might call it a hack? 😁 👍 )